src/DNA/native/src/Heap.c

```diff
@@ -72,6 +72,8 @@ struct tHeapEntry_ {
 	// unused
 	U8 padding;
 
+	U32 totalSize;
```

Comment:
Is this just for speed? To avoid the cost of a `GetSize()` call? It uses another 32-bits for every heap entry, which isn't disastrous, but isn't great either.

Reply:
@chrisdunelm Thanks, I really appreciate your input. Yes, this is mainly for speed at the price of one more int per heap entry. It does help a little but it's not a huge win in release mode. I'll hide that behind ifdef so that it is configurable.

Reply:
@ncave Have you profiled this? I'm just curious what difference it makes?
Mind you, the whole heap code really needs a rewrite to reduce per-object memory overhead, and to remove the need for the binary tree of allocations. Not really thinking of a generational GC, that would probably add to much code. This was something I vaguely intended to do, but never got around to.
The current heap code was just the simplest thing to get GC working quickly.
The very initial implementation did no GC at all. It was beautifully fast, but ran out of memory rather too quickly.

Reply:
@chrisdunelm
Yes I have, it does make a difference in debug mode (obviously) and much smaller (but still measurable) difference in release mode. I've disabled it for now, but leaving it as configurable option.

The heap is great for what it is (non-generational) and it works well, as long as the alloc pressure is not too heavy, that's why I have increased the limits a bit. Obviously it would be nice to have a better one eventually, since it can eat a lot of the performance in heavy alloc scenarios.

LOL at the no GC allocator, seems to be a fad [lately](http://tooslowexception.com/zero-garbage-collector-for-net-core/).